### DISCUSSION TITLE
DAI 2024-2025 - Practical work 1 - Gianni Cecchetto and Nathan Tschantz

### DISCUSSION THREAD

**Discussion by GianniCecchetto** [heig-vd-dai-course/heig-vd-dai-course#370] [Show and tell]
During this practical work, we wanted to do a cli app that take a file path to an image and convert it to another version made with ASCII characters.

https://github.com/GianniCecchetto/dai-practical-work-1
3f97849fd4b7f0012b4f69de0a4b55a1475a68f3

**Comment by ludelafo** [comment]
It seems good to me but we would need a bit more details and/or options for the "conversion" to be enough for this practical work.

It can come later but some kind of customization (size, style, characters used, etc.) would be nice to see.

@GeraudSilvestri, do you agree?

**Reply by GeraudSilvestri** [reply]
I agree with what @ludelafo said.

An easy way to be more specific, would be to detail the different parameters the user can enter with the expected values.

**Comment by GianniCecchetto** [comment]
With the parameter --resolution=[0..100] in percentage, we can adjust the resolution of the initial image.  
With the parameter --encoding=ASCII or UTF8, we can decide if the image is written with ASCII character or block elements from UTF8 like this one ▓.

This is a small idea of what we could do. Let us know if we still don't have enough details.

**Reply by ludelafo** [reply]
Perfect, thank you! That's enough for me. :)

**Reply by GeraudSilvestri** [reply]
Good for me as well 👍🏻
